packages/engine/src/common/classes/GenerateMeshBVHWorker.ts

```diff
@@ -41,6 +41,11 @@ export class GenerateMeshBVHWorker {
         if (error) {
           reject(new Error(error))
         } else {
+          // we need to replace the arrays because they're neutered entirely by the
+          // webworker transfer.
+          geometry.setAttribute('position', new BufferAttribute(position, 3))
```

Comment:
i dont think we want to replace the geometry data when it comes back, this can cause problems with physics and other things that rely on the geometry. what is the reason it is needed to be returned back?